pkg/cmd/clients_test.go

```diff
@@ -544,3 +544,244 @@ func TestMobileClientsCmd_TestCreateClient(t *testing.T) {
 		})
 	}
 }
+
+func TestMobileClientsCmd_SetClientValueFromJsonCmd(t *testing.T) {
+	cases := []struct {
+		Name         string
+		ClientName   string
+		MobileClient func() mc.Interface
+		ExpectError  bool
+		ErrorPattern string
+		ExpectUsage  bool
+		PatchFlag    string
+		Flags        []string
+		Validate     func(t *testing.T, c *v1alpha1.MobileClient)
+	}{
+		{
+			Name: "Test set with empty data",
+			MobileClient: func() mc.Interface {
+				fkMc := &mcFake.Clientset{}
+				fkMc.AddReactor("patch", "mobileclients", func(action kt.Action) (handled bool, ret runtime.Object, err error) {
+					return true, &v1alpha1.MobileClient{
+						Spec: v1alpha1.MobileClientSpec{
+							Name: "myapp",
+						},
+					}, nil
+				})
+				return fkMc
+			},
+			ClientName: "myapp",
+			PatchFlag:  "--patch={}",
+			Flags:      []string{"--namespace=myproject", "-o=json"},
+			Validate: func(t *testing.T, client *v1alpha1.MobileClient) {
+				if nil == client {
+					t.Fatal("expected a mobile client but got nil")
+				}
+				if client.Spec.Name != "myapp" {
+					t.Fatalf("expected an app with name %s but got %s ", "myapp", client.Spec.Name)
+				}
+			},
+		},
+		{
+			Name: "Test set with name data",
+			MobileClient: func() mc.Interface {
+				fkMc := &mcFake.Clientset{}
+				fkMc.AddReactor("patch", "mobileclients", func(action kt.Action) (handled bool, ret runtime.Object, err error) {
+					return true, &v1alpha1.MobileClient{
+						Spec: v1alpha1.MobileClientSpec{
+							Name: "my-new-app",
+						},
+					}, nil
+				})
+				return fkMc
+			},
+			ClientName: "myapp",
+			PatchFlag:  "--patch={\"spec\": {\"name\": \"my-new-name\"}}",
+			Flags:      []string{"--namespace=myproject", "-o=json"},
+			Validate: func(t *testing.T, client *v1alpha1.MobileClient) {
+				if nil == client {
+					t.Fatal("expected a mobile client but got nil")
+				}
+				if client.Spec.Name != "my-new-app" {
+					t.Fatalf("expected an app with name %s but got %s ", "myapp", client.Spec.Name)
+				}
+			},
+		},
+		{
+			Name: "Test set with name data",
```

Comment:
this test has the same name as the previous test and appears to be doing the same thing?